pkg/bigquery/config/definitions.json

```diff
@@ -0,0 +1,206 @@
+[
+  {
+    "uid": "e2ffe076-ab2c-4e5e-9587-a613a6b1c146",
+    "id": "data-bigquery",
+    "title": "BigQuery",
+    "documentation_url": "https://www.instill.tech/docs/vdp/data-connectors/bigquery",
+    "icon": "bigquery.svg",
+    "icon_url": "",
+    "public": true,
+    "custom": false,
+    "tombstone": false,
+    "spec": {
+      "resource_specification": {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "title": "BigQuery Connector Spec",
+        "type": "object",
+        "required": [
+          "json_key",
+          "project_id",
+          "dataset_id",
+          "table_name",
+          "schema"
+        ],
+        "additionalProperties": false,
+        "properties": {
+          "json_key": {
+            "title": "JSON Key File contents",
+            "description": "Contents of the JSON key file with access to the bucket.",
+            "type": "string",
+            "credential_field": true
+          },
+          "project_id": {
+            "title": "BigQuery Project ID",
+            "description": "Fill in your BigQuery Project ID.",
+            "type": "string"
+          },
+          "dataset_id": {
+            "title": "BigQuery Dataset ID",
+            "description": "Fill in your BigQuery Dataset ID.",
+            "type": "string"
+          },
+          "table_name": {
+            "title": "BigQuery Table Name",
+            "description": "Fill in your BigQuery Table Name.",
+            "type": "string"
+          },
+          "schema": {
```

Comment:
@praharshjain
The schema here is still so massive.
I'm thinking can we directly go to our ultimate goal, retrieving the schema from BiqQuery server?

Reply:
updated